Gems/O3DEMaterialEditor/Code/CMakeLists.txt

```diff
@@ -0,0 +1,86 @@
+
+if(NOT PAL_TRAIT_BUILD_HOST_TOOLS)
+    return()
+endif()
+
+# Currently we are in the Code folder: ${CMAKE_CURRENT_LIST_DIR}
+# Get the platform specific folder ${pal_dir} for the current folder: ${CMAKE_CURRENT_LIST_DIR}/Platform/${PAL_PLATFORM_NAME}
+# Note: o3de_pal_dir will take care of the details for us, as this may be a restricted platform
+#       in which case it will see if that platform is present here or in the restricted folder.
+#       i.e. It could here in our gem : Gems/O3DEMaterialEditor/Code/Platform/<platorm_name>  or
+#            <restricted_folder>/<platform_name>/Gems/O3DEMaterialEditor/Code
+o3de_pal_dir(pal_dir ${CMAKE_CURRENT_LIST_DIR}/Platform/${PAL_PLATFORM_NAME} ${o3de_gem_restricted_path} ${o3de_gem_path} ${o3de_gem_name})
+
+# Now that we have the platform abstraction layer (PAL) folder for this folder, thats where we will find the
+# traits for this platform. Traits for a platform are defines for things like whether or not something in this gem
+# is supported by this platform.
+include(${pal_dir}/PAL_${PAL_PLATFORM_NAME_LOWERCASE}.cmake)
+
+ly_add_target(
+    NAME O3DEMaterialEditor.Static STATIC
```

Comment:
I call the new gem, its targets, namespace and classes O3DEMaterialEditor just to make sure it doesn't collide with the ones from the stand alone MaterialEditor ones. It also makes it clearer for the review.